✨ Feat: #29 RecordDetailBottomSheet 구현

NOP-YA opened the pull request:
<!--
🙏 PR 제목 컨벤션 (Gitmoji + 타입 + 이슈 번호 + 작업 요약)
예시: ✨ Feat: #167 예약 취소 구현
※ PR 생성 시 Assignees 및 Labels 설정도 잊지 마세요!
-->

## ✨ What’s this PR?
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- Closes #29 
- Closes #36 
---

### 🧶 주요 변경 내용 (Summary)
<!-- 이번 PR에서 작업한 핵심 변경 사항을 작성해주세요 -->

1. RecordDetailBottomSheet 기본 구조 구현
     - 하프 모달 형태의 RecordDetailBottomSheet 레이아웃 구성
     - 제목, 날짜, 사진, 꽃 이름/꽃말 영역 기본 UI 구현
     - PresentationDetent에 따른 sheet 크기 조정
  
2. 제목 입력 및 편집 UX 개선
     - 제목 수정 버튼 및 편집 모드 추가
     - 키보드 팝업 시 레이아웃 밀림 문제 해결 (ScrollView + Geometry 적용)
     - 입력 중 모달 닫을 경우 자동 저장 처리

3. Menu 구성에 따른 버튼 수정
     - 피그마내에 있는 menu식 버튼으로 변경
     - 수정 완료 버튼 UI 반영
   
4. RecordDetailViewModel 구조 개선
     - 향후 UseCase에서 Entity Mapper가 쉽게 이루어지도록 데이터 구조체로 묶음

---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<!-- 이미지 크기 조절 예시: <img width="300" alt="설명" src="링크"> -->

<img width="250" alt="IMG_0001" src="https://github.com/user-attachments/assets/f8097529-0119-4db5-8459-22421c2cc795" />
<img width="250" alt="IMG_0004" src="https://github.com/user-attachments/assets/7117601f-31a6-435d-9417-16b6f2b2a1c8" />
<img width="250" alt="IMG_0005" src="https://github.com/user-attachments/assets/22d4612f-d0b3-400a-b684-de5097120b5a" />
<img width="250" alt="IMG_0006" src="https://github.com/user-attachments/assets/14151c95-ab4e-43cf-a93a-298d57677dee" />

---

### 🧪 테스트 / 검증 내역
<!-- 동작 확인 여부나 시나리오 테스트 내용을 간단히 써주세요 -->

- [x] UI 정상 동작 확인
- [x] iPhone 15, iOS 17.4 환경에서 정상 동작
- [ ] 다크모드 테스트는 이후 이슈로 분리 예정 (#000)

---

### 💬 기타 공유 사항
<!-- 리뷰어가 참고하면 좋을 정보, 고민했던 지점 등을 적어주세요 -->

- issue #30 은 구현이 안됐습니다. 디자인 어딨죠.. 
- 나중에 Data 끌어올 때가 좀 걱정되네요. 나름 고려한다고 하긴 했는데, 코드가 생각보다 복잡해져서

---

### 🙇🏻‍♀️ 리뷰 가이드 (선택)
<!-- 리뷰어가 중점적으로 보면 좋을 포인트가 있다면 알려주세요 -->

- "RecordViewModel.swift" 로직 처리 방식

NOP-YA committed:
✨ Feat. RecordDetailBottomSheet 기본 구조 구현

Why: 선택된 오브젝트의 상세 정보를 확인할 수 있는 모달 컴포넌트

How:
- RecordDetailBottomSheet 기본 레이아웃 및 뷰 구조 설계
- 제목, 날짜, 사진, 꽃 이름/꽃말 등의 정보 영역 구성
- 드래그 인터랙션 기반 하프모달 구조 적용

Tag: #feat #ui #bottomsheet #swiftui #record

NOP-YA committed:
✨ Feat. 제목 수정 기능 및 UX 개선 처리

Why:
- RecordDetailBottomSheet 내에 수정 기능을 추가하기 위함

How:
- 수정 버튼 및 편집 모드 상태 플래그 추가
- 편집 모드일 때 title 텍스트 수정 가능하도록 처리
- 키보드 올라올 때 레이아웃이 밀리는 문제를 ScrollView + Geometry로 해결
- 사용자가 입력 도중 모달을 닫을 경우, 자동으로 텍스트 저장되도록 개선

Tag: #feat #ui #keyboard #editmode #swiftui

freeskyES commented on Packages/Features/Sources/Features/Record/RecordDetail/RecordDetailViewModel.swift:
```diff
@@ -0,0 +1,113 @@
+    }
+    
+    func deleteImage(_ image: UIImage) {
+        withAnimation {
```
역시 헨리~ 구현이 빠르시군요. 잘 봤습니다! 

약간의 코멘트를 드려보면, 뷰모델안에서 사용했던 withAnimation을 view 로 옮겨보면 어떨까요?

```swift
// View
withAnimation {
    viewModel.deleteImage(image)
}
```

```swift
// ViewModel
func deleteImage(_ image: UIImage) {
    selectedImages.removeAll { $0 == image }
}

freeskyES commented on Packages/Features/Sources/Features/Record/RecordDetail/RecordDetailViewModel.swift:
```diff
@@ -0,0 +1,113 @@
+    @Published var title: String = ""
+    @Published var selectedImages: [UIImage] = []
+    
+    @Published var isEditing: Bool = false
```
현재 ViewModel 내 상태들을 각각 개별 `@Published`로 관리해주셨는데요,  
화면이 간단할 땐 이 구조도 깔끔하지만,
편집 상태 관리, 변경 감지 등을 고려하면 상태를 하나의 구조체로 묶는 것도 추천드립니다

예를 들어:
1. 구조체로 묶기
```swift
struct RecordDetail {
    var flowerName: String
    var flowerMeaning: String
    var location: String
    var date: String
    var title: String
    var selectedImages: [UIImage]
}
```

```swift
@Published var detail: RecordDetail         // 원본 (서버에서 받아온 or 저장된 값)
@Published var editedDetail: RecordDetail?  // 편집 중 값
@Published var isEditing: Bool = false
```

이렇게 구성하면, 이후 UseCase에서 받아온 Entity를 RecordDetail로 쉽게 매핑할 수 있고,
editedDetail과 detail을 비교하거나 롤백하는 로직도 명확하게 구성할 수 있습니다!

뷰에선 구현해주신 것 처럼 뷰모델에서 가져와 수정여부에 따라 다르게 보여줄 수 있어요!
```swift
if viewModel.isEditing {
    TextField("제목", text: ...)
} else {
    Text(viewModel.detail.title)
}
```

---

좀 더 욕심이 나신다면, 다음 스텝도 있습니다!

2. **편집 여부까지 묶기**

```swift
struct RecordDetailState {
    var record: RecordDetail
    var isEditing: Bool
}
@Published var state: RecordDetailState
```

3. 복잡한것 같으시면 더 간단하게,  필드 + isEditing만 가볍게 묶기

```swift
struct EditableRecordDetail {
    var title: String
    var ...: ...
    var isEditing: Bool
}

NOP-YA replied:
리뷰 감사합니다~ 플로우가 수정될 것 같다고 그래서 해당 내용 확인 하고 이것도 반영해둘게요.

NOP-YA committed:
♻️ Refactor. 상세 모달 플로우 피그마 반영 및 UI 정리

Why:
- 수정 플로우를 유연하게 구성하고, 향후 데이터 주입에 대비한 뷰와 뷰모델 역할 분리를 위해

How:
- 기존 수정 버튼을 Menu 방식으로 변경
- edit 모드에서 “수정 완료” 버튼을 별도 구성해 상태 명확화
- 사용자가 편집 도중 detent를 내리면 자동 저장되도록 개선
- 뷰모델 내 초기 구조 정리 및 데이터 주입 고려한 재구성
- 애니메이션 처리 코드를 뷰로 분리하여 MV 구조 명확화

Tag: #refactor #editmode #viewmodel #ui #swiftui
